[FIX] base - Fixed the issue of date widget in Qweb report which did not consider timezone.

odooaktiv opened the pull request:
**Description of the issue/feature this PR addresses:**
When in t-options widget "date" was used the date was not converted to correct timezone of the user.

**Current behavior before PR:**
1. In the Qweb date widget py logic, it used format_date() function which didn't considered the user timezone.
2. There was not use of 'tz_name' attribute inside the py logic.

**Desired behavior after PR is merged:**
Taking reference from "datetime" widget of Qweb report,
1. Used pycompat.to_text() method to convert to the correct timezone.
2. Timezone is taken from "tz_name" or "tz" from context.
3. date_format is taken from get_lang(self.env)

Fixes #79890
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/80168).

igallart commented:
I have same error in custom reports, and this solution has solved it for me.

igallart commented:
Related with https://github.com/odoo/odoo/issues/55827

odooaktiv commented:
@seb-odoo  Hey Sebastien, Can you please help review this PR?

seb-odoo commented:
> @seb-odoo Hey Sebastien, Can you please help review this PR?

Hello, I don't really know this part, I can't help much here.

odooaktiv commented:
@tde-banana-odoo Hey Thibault, Can you please help review this PR?

tde-banana-odoo commented:
Hello,

Thanks for the PR, seems interesting. I think it could be better to improve the ``date_format`` from tools and use it instead of copy-pasting most of the code. Indeed it is better to have a single well documented tool method than having two methods doing about the same but not totally. Could you give it a look ?

Cheers !

odooaktiv commented:
@tde-banana-odoo Sure Thibault, Thanks for your feedback. I will check `format_date` from tools tomorrow and see what can be changed there to better manage this.

odooaktiv commented:
@tde-banana-odoo Hey Thibault, I have added a new commit and updated the logic on the format_date() function of misc tools which now handles the timezone conversion. I have also updated the title and description of this PR represents the changes in the fix. Can you please review this PR?

odooaktiv commented:
@tde-banana-odoo Hey Thibault, Can you please review the PR?

tde-banana-odoo commented on odoo/tools/misc.py:
```diff
@@ -1301,7 +1301,11 @@ def format_date(env, value, lang_code=False, date_format=False):
 
-    return babel.dates.format_date(value, format=date_format, locale=locale)
+    tzinfo = None
+    if env.context.get("tz"):
```
Why not adding a tz=False additional parameter (suitable for stable policy as overrides / existing calls are still ok) to the method and calling it accordingly ?

cvinh commented:
Hello
This PR has fixed the issue in v14 aswell
Thank you
